From: Bart van Strien <bart.bes+projects@gmail.com>
Date: Wed, 3 May 2017 13:06:10 +0200
Subject: [PATCH] Fix luasocket compatibility with luajit 2.1.0 beta 3 (fix
 #1277)

Note that luasocket 3.0 (in minor) targets lua 5.1, so it does not rely on a
5.0 compatibility feature, and does not require this patch. Since a 0.10.3
release seems unlikely at this point, this commit is mostly targeted at future
packagers of older love versions.
---
 src/libraries/luasocket/libluasocket/lua.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/libraries/luasocket/libluasocket/lua.h b/src/libraries/luasocket/libluasocket/lua.h
index a280df84..c4215ecb 100644
--- a/src/libraries/luasocket/libluasocket/lua.h
+++ b/src/libraries/luasocket/libluasocket/lua.h
@@ -6,9 +6,12 @@
 #include <lualib.h>
 #include <lauxlib.h>
 
-#if LUA_VERSION_NUM > 501
+#if LUA_VERSION_NUM >= 501
+#	undef luaL_reg
 #	define luaL_reg luaL_Reg
+#	undef luaL_putchar
 #	define luaL_putchar(B, c) luaL_addchar(B, c)
+#	undef luaL_typerror
 #	define luaL_typerror(L, n, t) luax_typerror(L, n, t)
 
 extern int luax_typerror(lua_State *L, int narg, const char *type);
-- 
2.39.1

